credentials/insecure/insecure.go

```diff
@@ -70,3 +70,29 @@ type info struct {
 func (info) AuthType() string {
 	return "insecure"
 }
+
+// insecureBundle implements an insecure bundle.
+// An insecure bundle provides a thin wrapper around insecureTC to support
+// the credentials.Bundle interface.
+type insecureBundle struct{}
+
+// NewBundle returns a bundle with disabled transport security and no per rpc credential.
+func NewBundle() credentials.Bundle {
+	return insecureBundle{}
+}
+
+// NewWithMode returns a new insecure Bundle. The mode is ignored.
+func (insecureBundle) NewWithMode(_ string) (credentials.Bundle, error) {
+	return insecureBundle{}, nil
+}
+
+// PerRPCCredentials returns an nil implementation as insecure
+// bundle does not support a per rpc credential.
+func (insecureBundle) PerRPCCredentials() credentials.PerRPCCredentials {
+	return nil
+}
+
+// TransportCredentials returns the underlying insecure transport credential.
+func (insecureBundle) TransportCredentials() credentials.TransportCredentials {
+	return NewCredentials()
```

Comment:
Is there a reason for return a new instance of `insecureTC` for every call to `TransportCredentials`? I understand both `insecureBundle` and `insecureTC` and empty structs. But I would assume returning a new one for every call would still not be free.

Reply:
I thought we could store an instance of `insecureTC` inside `insecureBundle`. But just realized that both use value receivers and not pointer receivers. So, that might not be useful.

Reply:
Changed it back to empty struct with NewCredentials() being called.